### ISSUE TITLE
reject after resolve?

### ISSUE THREAD

**Issue by srl295** [jyu213/ssh2-sftp-client#22]
Hi,
 in https://github.com/jyu213/ssh2-sftp-client/blob/master/src/index.js#L73
there is (in the `get()` function) 
```js
                let stream = sftp.createReadStream(path, options);
                stream.on('error', reject);
                resolve(stream);
```

however, the `reject()` is not called (for example on a missing file) because `resolve()` was already called. 

I'm not sure, but perhaps it should be something like:   `stream.on('readable', ()=>resolve(stream))` so that it waits until the stream is readable? Still learning about streams.

**Comment by srl295** [comment]
So I worked past this like so.

I had this:
```js
sftp.get(path)
.then(doSomethingWithStream)
…
```

but it would never continue if the file was not found. 

I changed it to:


```js
sftp.get(path)
.then((stream) => new Promise((resolve, reject) => {
   stream.on('error', reject);
   doSomethingWithStream(stream)  // 
   .then(resolve, reject);
})
.then(…) // worked! Continue chain…
.catch((e) =>{
   // error; could be file not found or something else
})
```

I tried `stream.on('readable', …)` but that didn't work— listening to that event caused the stream not to be usable later on somehow.

Thanks!

**Comment by jyu213** [comment]
@srl295 thanks for your advise.I will fix it later. 

how about use `stream.on('end', ...)`?

**Comment by cjnqt** [comment]
Any news on this?

**Comment by jyu213** [comment]
update new version 2.0.1

**Comment by waldyrious** [comment]
For future reference, the fix was done in ab32b227f.
